identity/key.go

```diff
@@ -1,18 +1,73 @@
 package identity
 
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/openpgp/armor"
+	"golang.org/x/crypto/openpgp/packet"
+)
+
 type Key struct {
-	// The GPG fingerprint of the key
-	Fingerprint string `json:"fingerprint"`
-	PubKey      string `json:"pub_key"`
+	// PubKey is the armored PGP public key.
+	ArmoredPublicKey string `json:"pub_key"`
+	// ExpireUnixTime is when the key becomes invalid.
+	ExpireUnixTime int64 `json:"expire_unix_time"`
```

Comment:
It might makes sense to have an expire time but `Identity.ValidKeyAtTime` should then be updated to filter the expired one. Using a timestamp also make sense but it open a whole can of problems : what happen if clocks are incorrect ? There is a lot of failure mode here. Do you **really** need that concept ? What for ?

Reply:
Good question. I'm not sure what's the usefulness of the expiring time.

The `ValidKeysAtTime(time lamport.Time)` method gets only a `lamport.Time` value, so it can only be used for validating the identities. 

For validating a commit we check differently. Let's say a key is removed. We have a commit signed with the said removed key. How do we check if the key was valid at the time the commit was made (committer time)? We have the commit unix time, and the commit unix times of the two operations introducing and removing the key.

The point is that the expiring time would be used in the commit validation mechanism, not in the identity validation one, so we can add it there in case somebody needs it. 

Makes sense?

Reply:
Are you talking about commits *made by git-bug*, ie operations, or actual code commits ?

Reply:
If you are talking of validating operations, you don't need any real time. Each operation has a Lamport time that you can use to query which keys are valid at that logical time. As the identity is a chain of version, we keep around all the keys even when removed.

Now, for validating code commits ... it's another story. Without adding an extra expiry time to a key, we can use the existing unixtime of the version the keys are attached to to derive a validity period. This way we stay close to the normal git-bug way and not open a window for a lot of new failure scenario. It would "just" require a new `Identity.ValidKeysAtUnixTime()` function.

As we can't trust time, we are subject to false positive / false negative. It might be fine but rules need to be decided for what happen in those cases. One possible way to make this process a bit more robust could be to check that the unix time of both identity's version and code commit are monotonic (ie, stated time and data location match in ordering). This would at least bring us closer to the full logical ordering of the Lamport time and it's nice properties. Problem is, it's not a complete guarantee and if we detect such a problem, the best we can do is report that to the user as "something's weird I don't know".

Reply:
Ok, let's forget about expiration times, and use only the unix time of the commit which removed the key.